tests/performance/performance_test_driver.h

```diff
@@ -0,0 +1,120 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_performance_test_driver_h
+#define dealii_performance_test_driver_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+#include <string>
+#include <vector>
+
+
+/**
+ * An enum class that describes the environment in which performance tests
+ * are running.
+ */
+enum class TestingEnvironment
+{
+  /**
+   * A lightweight mobile laptop with at least 2 physical cores and at
+   * least 8 GB of main memory.
+   */
+  light,
+  /**
+   * A medium sized work station with at least 8 physical cores and at
+   * least 32 GB of main memory.
+   */
+  medium,
+  /**
+   * A heavyweight compute node with at least 32 physical cores and at
+   * least 128 GB of main memory.
+   */
+  heavy,
+};
```

Comment:
@peterrum 

As we discussed, the configuration of a compute node/cluster/workstation should be set via `TEST_MPI_RANKS_LIMIT` and `TEST_THREAD_LIMIT`, and tests that should run on the entire node/machine should then be declared via `mpirun=max` and `threads=max`.

However, this still leaves the issue that a performance test doesn't quite know how heavy the work load can be. I suggest we introduce a simple `TestingEnvironment` enum class for this reason describing resource limits. After all, even if I tell you what amount of memory or CPU time a test is allowed to use - it is really hard to translate this information into a precise choice of refinement parameters. So in my opinion having three choices `light`, `medium`, `heavy`, is good enough.